com.chartboost.helium/Android~/helium-android-bridge/src/main/kotlin/com/chartboost/heliumsdk/unity/HeliumUnityBridge.kt

```diff
@@ -0,0 +1,390 @@
+import com.chartboost.heliumsdk.unity.HeliumEventProcessor.HeliumEventConsumer
+import com.chartboost.heliumsdk.unity.HeliumEventProcessor.HeliumRewardEventConsumer
+import com.chartboost.heliumsdk.unity.HeliumEventProcessor.serializeHeliumBidEvent
+import com.chartboost.heliumsdk.unity.HeliumEventProcessor.serializeHeliumEvent
+import com.chartboost.heliumsdk.unity.HeliumEventProcessor.serializeHeliumRewardEvent
+import com.chartboost.heliumsdk.unity.HeliumEventProcessor.serializePlacementILRDData
+import com.chartboost.heliumsdk.unity.HeliumUnityAdWrapper.Companion.wrap
+import com.unity3d.player.UnityPlayer
+
+@Suppress("NAME_SHADOWING")
+class HeliumUnityBridge {
+    private lateinit var _activity: Activity
+    private var lifeCycleEventListener: ILifeCycleEventListener? = null
+    private var bannerEventsListener: IBannerEventListener? = null
+    private var interstitialEventsListener: IInterstitialEventListener? = null
+    private var rewardedEventListener: IRewardedEventListener? = null
+    private var ilrdObserver: HeliumIlrdObserver? = null
+    fun setupEventListeners(
+        lifeCycleListener: ILifeCycleEventListener?,
+        interstitialListener: IInterstitialEventListener?,
+        rewardedListener: IRewardedEventListener?,
+        bannerListener: IBannerEventListener?
+    ) {
+        lifeCycleEventListener = lifeCycleListener
+        interstitialEventsListener = interstitialListener
+        rewardedEventListener = rewardedListener
+        bannerEventsListener = bannerListener
+    }
+
+    fun setSubjectToCoppa(isSubject: Boolean) {
+        HeliumSdk.setSubjectToCoppa(isSubject)
+    }
+
+    fun setSubjectToGDPR(isSubject: Boolean) {
+        HeliumSdk.setSubjectToGDPR(isSubject)
+    }
+
+    fun setCCPAConsent(hasGivenConsent: Boolean) {
+        HeliumSdk.setCCPAConsent(hasGivenConsent)
+    }
+
+    fun setUserHasGivenConsent(hasGivenConsent: Boolean) {
+        HeliumSdk.setUserHasGivenConsent(hasGivenConsent)
+    }
+
+    var userIdentifier: String?
+        get() = HeliumSdk.getUserIdentifier()
+        set(userIdentifier) {
+            HeliumSdk.setUserIdentifier(userIdentifier)
+        }
+
+    fun onBackPressed(): Boolean {
+        return HeliumSdk.onBackPressed()
+    }
+
+    fun start(appId: String?, appSignature: String?, unityVersion: String?) {
+        _activity = UnityPlayer.currentActivity
+
+        ilrdObserver = object : HeliumIlrdObserver {
+            override fun onImpression(impData: HeliumImpressionData) {
+                val json = serializePlacementILRDData(impData.placementId, impData.ilrdInfo)
+                lifeCycleEventListener!!.DidReceiveILRD(json)
+            }
+        }
+        runTaskOnUiThread {
+            // This call initializes the Helium SDK. This might change in the future with two ID parameters and we'll get rid of the logControllerListener
+            HeliumSdk.start(_activity, appId!!, appSignature!!) { error: Error? ->
```

Comment:
please don't use `!!`
make it null safe.

Reply:
Yes, this should not be nullable